ironfish-cli/src/commands/ceremony.ts

```diff
@@ -0,0 +1,170 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { contribute } from '@ironfish/rust-nodejs'
+import { ErrorUtils, PromiseUtils } from '@ironfish/sdk'
+import { CliUx, Flags } from '@oclif/core'
+import axios from 'axios'
+import fsAsync from 'fs/promises'
+import path from 'path'
+import { pipeline } from 'stream/promises'
+import { IronfishCommand } from '../command'
+import { DataDirFlag, DataDirFlagKey, VerboseFlag, VerboseFlagKey } from '../flags'
+import { CeremonyClient } from '../trusted-setup/client'
+
+export default class Ceremony extends IronfishCommand {
+  static description = 'Contribute randomness to the Iron Fish trusted setup'
+
+  static flags = {
+    [VerboseFlagKey]: VerboseFlag,
+    [DataDirFlagKey]: DataDirFlag,
+    host: Flags.string({
+      parse: (input: string) => Promise.resolve(input.trim()),
+      default: '127.0.0.1',
+      description: 'Host address of the ceremony coordination server',
+    }),
+    port: Flags.integer({
+      default: 9040,
+      description: 'Port of the ceremony coordination server',
+    }),
+  }
+
+  async start(): Promise<void> {
+    const { flags } = await this.parse(Ceremony)
+    const { host, port } = flags
+
+    // Pre-make the temp directory to check for access
+    const tempDir = this.sdk.config.tempDir
+    await fsAsync.mkdir(tempDir, { recursive: true })
+
+    const inputPath = path.join(tempDir, 'params')
+    const outputPath = path.join(tempDir, 'newParams')
+
+    let localHash: string | null = null
+
+    // Prompt for randomness
+    let randomness: string | null = await CliUx.ux.prompt(
```

Comment:
Need to create this blog post and insert

Reply:
I'm going to remove that for now and add an airtable task to re-add it once the blog post is up.